config/config.go

```diff
@@ -635,6 +635,8 @@ type ScrapeConfig struct {
 	ScrapeProtocols []ScrapeProtocol `yaml:"scrape_protocols,omitempty"`
 	// Whether to scrape a classic histogram that is also exposed as a native histogram.
```

Comment:
I assume for case of `ScrapeClassicHistograms=true` and `ConvertClassicHistograms=true` this will create both nh and nhcb?

Reply:
yes, that's the idea

Reply:
hm, not really -- from your code it looks like `ScrapeClassicHistograms` is setting `keepClassicHistograms`, so if ConvertClassicHistograms && ScrapeClassicHistograms == true we have both classic histograms, nhcb and nh.

Why? Are we not overcomplicating?

Also name of this option is super confusing, it looks like you have to manually decide to keep classic histograms - but I guess we cannot change it (or maybe we can, for 3.0?)

Reply:
hmm, I'll add a new test, we don't intend to scrape nhcb and nh at the same time ever

Reply:
I've made tables back in the day, time to encode in tests: https://github.com/prometheus/prometheus/issues/13532

Reply:
scrape/scrape_test.go implements the table in #13532 and the end result is correct. However the parser does return both exponential and custom bucket histograms. That are both added to the head appender , but head.Commit() will throw away the custom bucket histogram due to duplicate sample for timestamp: 
![image](https://github.com/user-attachments/assets/3c2e6bc9-a2f7-4f0d-b8f3-fd42903635e1)

Reply:
Added unit test to have  a TODO: https://github.com/prometheus/prometheus/pull/14978/commits/5ee0980cd1d50881f9693a85c486f25bca2976da

Reply:
What's the next step here? Should we add comment about this behaviour? Essentially on `AlwaysScrapeClassicHistograms` we could say `unless ConvertClassicHistogramsToNHCB is true`?

Reply:
`AlwaysScrapeClassicHistograms`  is not violated if `ConvertClassicHistogramsToNHCB` is true. See this case in the scrape_test.go:
```
if metricsText.hasClassic {
				switch {
				case tc.alwaysScrapeClassicHistograms && tc.convertClassicHistToNHCB:
					expectedClassicHistCount = 1
					expectedNativeHistCount = 1
					expectCustomBuckets = true
```

Reply:
I think the next step is to make the scrape_test.go addition (`TestConvertClassicHistogramsToNHCB`) less complicated - easier to understand. Maybe even in this PR?